src/main/java/duke/command/DeadlineCommand.java

```diff
@@ -0,0 +1,55 @@
+package duke.command;
+
+import java.io.IOException;
+import java.time.LocalDate;
+
+import duke.exception.DukeException;
+import duke.storage.Storage;
+import duke.task.Deadline;
+import duke.tasklist.TaskList;
+import duke.ui.Ui;
+
+/**
+ * The DeadlineCommand class encapsulates the execution of the deadline command from the user.
+ */
+public class DeadlineCommand extends Command {
+    /** The deadline command inputted by the user. */
+    String fullCommand;
+
+    /**
+     * Constructor to intialise a DeadlineCommand.
+     * @param fullCommand The deadline command inputted by the user.
+     */
+    public DeadlineCommand(String fullCommand) {
+        this.fullCommand = fullCommand;
+    }
+
+    /**
+     * Executes the response to the deadline command from the user.
+     * @param storage The storage Duke uses to save and load the tasklist from.
+     * @param taskList The list of tasks Duke needs to execute on.
+     * @param ui The Ui Duke utilises to interact with the user.
+     */
+    @Override
+    public void execute(Storage storage, TaskList taskList, Ui ui) throws IOException, DukeException {
+        try {
+            if (fullCommand.equals("deadline")) {
+                throw new DukeException("☹ OOPS!!! The description of a deadline cannot be empty.");
+            }
+
+            String deadlineDescription = fullCommand.split(" ", 2)[1];
+            int pos = deadlineDescription.indexOf("/");
+            String description = deadlineDescription.substring(0, pos - 1);
+            String by = deadlineDescription.substring(pos + 4);
```

Comment:
Would it be better to name this as 'duedate' instead? If I am not wrong, the variable represents a date/noun, while 'by' might not be the most suitable name in this case. Similarly, I have noticed that you have used this name at some other areas as well.